sui_programmability/examples/nfts/sources/Chat.move

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module NFTs::Chat {
+    use Sui::ID::{Self, ID, VersionedID};
+    use Std::ASCII::{Self, String};
+    use Std::Option::Option;
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+
+    /// Max text length.
+    const MAX_TEXT_LENGTH: u64 = 512;
+
+    /// Text size overflow.
+    const ETextOverflow: u64 = 0;
+
+    /// Sui Chat NFT (i.e., a post, retweet, like, chat message etc).
+    struct Chat has key, store {
+        id: VersionedID,
+        // The ID of the chat app.
+        app_id: ID,
+        // Post's text.
+        text: String,
+        // Set if referencing an another object (i.e., due to a Like, Retweet, Reply etc).
+        // We allow referencing any object type, not ony Chat NFTs.
+        ref_id: Option<ID>,
+        // app-specific metadata.
+        metadata: vector<u8>,
+    }
+
+    /// Simple Chat.text getter.
+    public fun text(chat: &Chat): String {
+        chat.text
+    }
+
+    /// Mint (post) a Chat object.
+    public(script) fun mint(
+            app_id: ID,
```

Comment:
Did this pass `sui-move test`? I thought we cannot pass `ID` to entry functions.

Reply:
Good point, I was expecting that our CI will capture that. Actually replaced all `ID`, `Option` and `String` with primitive types, and we expect application devs will work by "convention" (i.e. provide an `address` instead of object `ID`, and `vector<u8>` instead of `String`). Back to you @lxfind